exporter/prometheusexporter/collector.go

```diff
@@ -79,16 +81,41 @@ func (c *collector) metricName(namespace string, metric pdata.Metric) string {
 	return sanitize(metric.Name(), c.skipSanitizeLabel)
 }
 
-func (c *collector) getMetricMetadata(metric pdata.Metric, attributes pdata.Map) (*prometheus.Desc, []string) {
-	keys := make([]string, 0, attributes.Len())
-	values := make([]string, 0, attributes.Len())
+func (c *collector) getMetricMetadata(metric pdata.Metric, attributes pdata.Map, resourceAttrs pdata.Map) (*prometheus.Desc, []string) {
+	keys := make([]string, 0, attributes.Len()+2) // +2 for job and instance labels.
+	values := make([]string, 0, attributes.Len()+2)
 
 	attributes.Range(func(k string, v pdata.Value) bool {
 		keys = append(keys, sanitize(k, c.skipSanitizeLabel))
 		values = append(values, v.AsString())
 		return true
 	})
 
+	// Sometimes the map can be empty and lookups on an empty map panic.
+	if (resourceAttrs == pdata.Map{}) {
```

Comment:
I don't think we need to handle this case.  We don't do this for metric attributes, for example (we Range on it above).  Tests should use pdata.NewMap().

Reply:
Good catch, removing this check surfaced another bug 🙃 

I think it might be ready to go now!